# 🔍 Exploration et Audit des fichiers CSV bruts

## Installation

In [41]:
! pip install pandas 

Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


## Importation

In [318]:
import pandas as pd
import glob
import os

## 📊 Source 1 : Reddit WallStreetBets

In [178]:
df_redit = pd.read_csv("C:\\Users\\semy4\\OneDrive\\Bureau\\Fintech_project\\data\\raw\\reddit_stealth_data_massive.csv", sep=",")

In [179]:
df_redit

,Date_Publication,Ticker,Subreddit,Titre
0,2026-07-10,AAPL,wallstreetbets,Is it a sign?
1,2026-07-09,AAPL,wallstreetbets,"Hands down, MSFT is the worst fucking investme..."
2,2026-07-08,AAPL,wallstreetbets,Apple announces chip deal with Broadcom worth ...
3,2026-07-03,AAPL,wallstreetbets,Every dip is buying opportunity in Apple.
4,2026-06-30,AAPL,wallstreetbets,ASML is my first 200% gain ever. Bought during...
...,...,...,...,...
120,2026-03-27,JPM,wallstreetbets,Dow Jones & NASDAQ Composite close in -10% cor...
121,2026-02-22,JPM,wallstreetbets,The moment I buy a thousand dollars worth of puts
122,2026-01-31,JPM,wallstreetbets,A Timeline of Today's Silver Crash and its Ben...
123,2026-01-23,JPM,wallstreetbets,And we’re back 😏


In [180]:
df_redit.columns

Index(['Date_Publication', 'Ticker', 'Subreddit', 'Titre'], dtype='object')

In [166]:
df_redit.tail

<bound method NDFrame.tail of     Date_Publication Ticker       Subreddit  \
0         2026-07-10   AAPL  wallstreetbets   
1         2026-07-09   AAPL  wallstreetbets   
2         2026-07-08   AAPL  wallstreetbets   
3         2026-07-03   AAPL  wallstreetbets   
4         2026-06-30   AAPL  wallstreetbets   
..               ...    ...             ...   
120       2026-03-27    JPM  wallstreetbets   
121       2026-02-22    JPM  wallstreetbets   
122       2026-01-31    JPM  wallstreetbets   
123       2026-01-23    JPM  wallstreetbets   
124       2026-01-22    JPM  wallstreetbets   

                                                 Titre  
0                                        Is it a sign?  
1    Hands down, MSFT is the worst fucking investme...  
2    Apple announces chip deal with Broadcom worth ...  
3            Every dip is buying opportunity in Apple.  
4    ASML is my first 200% gain ever. Bought during...  
..                                                 ...  
120  D

In [148]:
df_redit.shape

(125, 4)

In [149]:
df_redit.dtypes

Date_Publication    object
Ticker              object
Subreddit           object
Titre               object
dtype: object

In [150]:
df_redit.isnull().sum()

Date_Publication    0
Ticker              0
Subreddit           0
Titre               0
dtype: int64

In [152]:
df_redit.nunique()

Date_Publication     57
Ticker                9
Subreddit             1
Titre               102
dtype: int64

In [181]:
nombre_doublons = df_redit.duplicated(subset=['Ticker', 'Date_Publication', 'Titre']).sum()
print(f"Nombre de doublons trouvés : {nombre_doublons}")

Nombre de doublons trouvés : 0


In [153]:
df_redit['Subreddit'].unique()

array(['wallstreetbets'], dtype=object)

In [154]:
df_redit['Ticker'].unique()

array(['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B',
       'JPM'], dtype=object)

### 🔴 Source 1 : Reddit - r/wallstreetbets (Audit des Données)

L'analyse exploratoire initiale de la première source de données textuelles met en évidence les caractéristiques structurelles suivantes :

*   **Volumétrie du Dataset :** Le fichier comporte **125 lignes** et **4 colonnes** (Shape: `(1961, 4)`).
*   **Types de Données :** L'ensemble des variables (`Date_publication`, `Ticker`, `Subreddit`, `Titre`) est typé en tant qu'**objets** (`dtype='object'`), ce qui correspond au format standard de Pandas pour les chaînes de caractères (strings).
*   **Qualité des Données :** Excellente intégrité globale. On note une **absence totale de valeurs manquantes** (`NaN = 0`) sur toutes les colonnes.
*   **Distribution des Variables Clés :**
    *   **Subreddit :** La source est parfaitement homogène avec une seule modalité unique : `['wallstreetbets']`.
    *   **Tickers :** Le périmètre d'étude cible rigoureusement les **9 actifs majeurs** configurés dans notre pipeline de collecte, sans aucune valeur aberrante : `['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B', 'JPM']`.

**Conclusion pour la standardisation :** Ce jeu de données est propre. La seule transformation nécessaire lors de la fusion sera l'alignement de la colonne temporelle (`Date_publication`) avec les formats standards de type Date, Subreddits et Ticker aussi .

## 📊 Source 2 : Reddit Stocks

In [184]:
df_redit_stocks = pd.read_csv("C:\\Users\\semy4\\OneDrive\\Bureau\\Fintech_project\\data\\raw\\reddit_stocks_data_massive.csv", sep=",")

In [156]:
df_redit_stocks.head()

,Date_Publication,Ticker,Subreddit,Titre
0,2026-07-10,AAPL,stocks,r/Stocks Daily Discussion & Fundamentals Frida...
1,2026-07-03,AAPL,stocks,r/Stocks Daily Discussion & Fundamentals Frida...
2,2026-06-26,AAPL,stocks,r/Stocks Daily Discussion & Fundamentals Frida...
3,2026-06-23,AAPL,stocks,Markets are not driven by intrinsic value
4,2026-06-19,AAPL,stocks,r/Stocks Daily Discussion & Fundamentals Frida...


In [185]:
df_redit_stocks.columns

Index(['Date_Publication', 'Ticker', 'Subreddit', 'Titre'], dtype='object')

In [186]:
df_redit_stocks.shape

(97, 4)

In [187]:
df_redit_stocks.dtypes

Date_Publication    object
Ticker              object
Subreddit           object
Titre               object
dtype: object

In [188]:
df_redit_stocks.isnull().sum()

Date_Publication    0
Ticker              0
Subreddit           0
Titre               0
dtype: int64

In [189]:
df_redit_stocks.nunique()

Date_Publication    44
Ticker               9
Subreddit            1
Titre               73
dtype: int64

In [190]:
nombre_doublons = df_redit_stocks.duplicated(subset=['Ticker', 'Date_Publication', 'Titre']).sum()
print(f"Nombre de doublons trouvés : {nombre_doublons}")

Nombre de doublons trouvés : 0


In [162]:
df_redit_stocks['Subreddit'].unique()

array(['stocks'], dtype=object)

In [163]:
df_redit_stocks['Ticker'].unique()

array(['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B',
       'JPM'], dtype=object)

### 🔴 Source 2 : Reddit - r/Stocks (Audit des Données)

L'analyse exploratoire initiale de la 2 éme  source de données textuelles met en évidence les caractéristiques structurelles suivantes :

*   **Volumétrie du Dataset :** Le fichier comporte **2264 lignes** et **4 colonnes** (Shape: `(2264, 4)`).
*   **Types de Données :** L'ensemble des variables (`Date_Publication`, `Ticker`, `Subreddit`, `Titre`) est typé en tant qu'**objets** (`dtype='object'`), ce qui correspond au format standard de Pandas pour les chaînes de caractères (strings).
*   **Qualité des Données :** Excellente intégrité globale. On note une **absence totale de valeurs manquantes** (`NaN = 0`) sur toutes les colonnes.
*   **Distribution des Variables Clés :**
    *   **Subreddit :** La source est parfaitement homogène avec une seule modalité unique : `['wallstreetbets']`.
    *   **Tickers :** Le périmètre d'étude cible rigoureusement les **9 actifs majeurs** configurés dans notre pipeline de collecte, sans aucune valeur aberrante : `['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B', 'JPM']`.

**Conclusion pour la standardisation :** Ce jeu de données est propre. La seule transformation nécessaire lors de la fusion sera l'alignement de la colonne temporelle (`Date_Publication') avec les formats standards de type Date, Subreddits et Ticker aussi .

## 📊 Source 3 : News API

In [198]:
df_news=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\raw\api_news_dataset_MAX.csv" , sep=",")

In [199]:
df_news.head()

,Date_Publication,Ticker,Source,Titre,Description,URL
0,2025-08-31,AAPL,Yahoo,California's Rep. Khanna Says Intel Stake Shou...,Democratic California Rep. Ro Khanna said fede...,https://finnhub.io/api/news?id=2a20d2f44b1cdb9...
1,2025-08-31,AAPL,Yahoo,August jobs report due out as Fed uncertainty ...,"A crucial August jobs report, drama at the Fed...",https://finnhub.io/api/news?id=3056cda71b6a153...
2,2025-08-31,AAPL,Yahoo,"What Bad Bunny, F1, and Katseye reveal about t...",The consumer appears to be highly engaged.,https://finnhub.io/api/news?id=adf84d45f17e891...
3,2025-08-31,AAPL,Yahoo,A New iPhone Is Coming. Should You Upgrade or ...,By improving sluggish performance or replacing...,https://finnhub.io/api/news?id=eb3a11dca0e20b7...
4,2025-08-31,AAPL,Yahoo,Could Buying O'Reilly Automotive Stock Today H...,Shares of this retailer are up an astonishing ...,https://finnhub.io/api/news?id=0aba0cff2b5c601...


In [200]:
df_news

,Date_Publication,Ticker,Source,Titre,Description,URL
0,2025-08-31,AAPL,Yahoo,California's Rep. Khanna Says Intel Stake Shou...,Democratic California Rep. Ro Khanna said fede...,https://finnhub.io/api/news?id=2a20d2f44b1cdb9...
1,2025-08-31,AAPL,Yahoo,August jobs report due out as Fed uncertainty ...,"A crucial August jobs report, drama at the Fed...",https://finnhub.io/api/news?id=3056cda71b6a153...
2,2025-08-31,AAPL,Yahoo,"What Bad Bunny, F1, and Katseye reveal about t...",The consumer appears to be highly engaged.,https://finnhub.io/api/news?id=adf84d45f17e891...
3,2025-08-31,AAPL,Yahoo,A New iPhone Is Coming. Should You Upgrade or ...,By improving sluggish performance or replacing...,https://finnhub.io/api/news?id=eb3a11dca0e20b7...
4,2025-08-31,AAPL,Yahoo,Could Buying O'Reilly Automotive Stock Today H...,Shares of this retailer are up an astonishing ...,https://finnhub.io/api/news?id=0aba0cff2b5c601...
...,...,...,...,...,...,...
23699,2026-07-01,JPM,Yahoo,Dimon says bull market is like a 'little tsuna...,The JPMorgan Chase CEO says he’s been caught o...,https://finnhub.io/api/news?id=74a3033d1c60764...
23700,2026-07-01,JPM,Yahoo,S&P Winning Streak for July at Risk With Volat...,(Bloomberg) -- July is a great time to buy sto...,https://finnhub.io/api/news?id=cb8ee519bc191bd...
23701,2026-07-01,JPM,Benzinga,"'JPMorgan AI Chief, Longtime Top Trader Exits ...",https://www.bloomberg.com/news/articles/2026-0...,https://finnhub.io/api/news?id=f3ffff991231a57...
23702,2026-07-01,JPM,Yahoo,GIX Appoints Sara Dillon as Chief Technology O...,Former NYSE and J.P. Morgan technology executi...,https://finnhub.io/api/news?id=4aec6dae688820d...


In [204]:
df_news.columns

Index(['Date_Publication', 'Ticker', 'Source', 'Titre', 'Description', 'URL'], dtype='object')

In [205]:
df_news.tail

<bound method NDFrame.tail of       Date_Publication Ticker    Source  \
0           2025-08-31   AAPL     Yahoo   
1           2025-08-31   AAPL     Yahoo   
2           2025-08-31   AAPL     Yahoo   
3           2025-08-31   AAPL     Yahoo   
4           2025-08-31   AAPL     Yahoo   
...                ...    ...       ...   
23699       2026-07-01    JPM     Yahoo   
23700       2026-07-01    JPM     Yahoo   
23701       2026-07-01    JPM  Benzinga   
23702       2026-07-01    JPM     Yahoo   
23703       2026-07-01    JPM     Yahoo   

                                                   Titre  \
0      California's Rep. Khanna Says Intel Stake Shou...   
1      August jobs report due out as Fed uncertainty ...   
2      What Bad Bunny, F1, and Katseye reveal about t...   
3      A New iPhone Is Coming. Should You Upgrade or ...   
4      Could Buying O'Reilly Automotive Stock Today H...   
...                                                  ...   
23699  Dimon says bull market is 

In [206]:
df_news.dtypes

Date_Publication    object
Ticker              object
Source              object
Titre               object
Description         object
URL                 object
dtype: object

In [76]:
df_news.shape

(2203, 6)

In [210]:
df_news.isnull().sum()

Date_Publication      0
Ticker                0
Source                0
Titre                 0
Description         413
URL                   0
dtype: int64

In [211]:
df_news.nunique()

Date_Publication      338
Ticker                  9
Source                  9
Titre               17850
Description         17161
URL                 18250
dtype: int64

### 📰 Source 3 : API Finnhub / News (Audit des Données)

L'analyse exploratoire de cette troisième source met en évidence une structure plus riche, mais également les premières anomalies de complétude à traiter :

*   **Volumétrie du Dataset :** Le fichier comporte **2 203 lignes** et **6 colonnes** (Shape: `(2203, 6)`).
*   **Types de Données :** Toutes les variables (`Date_Publication`, `Ticker`, `Source`, `Titre`, `Description`, `URL`) sont stockées sous forme d'**objets** (`dtype='object'`).
*   **Qualité des Données & Valeurs Manquantes :** 
    *   La colonne `Description` présente **12 valeurs manquantes** (`NaN`), ce qui nécessitera une stratégie d'imputation ou de suppression lors de la phase de nettoyage.
*   **Distribution et Cardinalité (Valeurs Uniques) :**
    *   **Date_Publication (89 uniques) :** L'échantillon couvre une période concentrée (environ 89 jours ou fenêtres temporelles distinctes).
    *   **Ticker (9 uniques) :** On retrouve parfaitement l'intégralité de notre panel de 9 entreprises cibles.
    *   **Source (8 uniques) :** Les articles proviennent de 8 agences de presse ou médias financiers différents, ce qui assure une bonne diversité éditoriale.
    *   **Titre (1742 uniques) & Description (1686 uniques) :** Le nombre de valeurs uniques est inférieur au nombre total de lignes (2 203). Cela indique la présence de **doublons textuels** (mêmes articles publiés plusieurs fois ou partagés entre différents tickers) qu'il faudra filtrer pour éviter de biaiser les scores de sentiment.
    *   **URL (1750 uniques) :** Confirme la présence d'articles dupliqués ou redondants à nettoyer.

**Conclusion pour la standardisation :** Ce dataset demande plus d'attention. Il faudra gérer les 12 lignes sans description, mais surtout appliquer un filtre de déduplication (`drop_duplicates`) sur les titres ou les URLs avant de procéder à la fusion, afin de ne pas sur-pondérer artificiellement certaines actualités répétitives.

## 📊 Source 4 : Rss_financiers 

In [215]:
df_rss=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\raw\rss_news_dataset_massive.csv" , sep=",")

In [216]:
df_rss

,Date_Publication,Ticker,Source,Titre,URL
0,2020-02-19,AAPL,The Motley Fool,Warren Buffett's Berkshire Hathaway Unloads 3....,https://news.google.com/rss/articles/CBMingFBV...
1,2020-02-02,AAPL,Barron's,Al Gore’s Apple Stock Is Now Worth $35 Million...,https://news.google.com/rss/articles/CBMikgFBV...
2,2020-02-10,AAPL,Investor's Business Daily,"S&P 500: What $10,000 Invested In FAANG Stocks...",https://news.google.com/rss/articles/CBMitwFBV...
3,2020-02-20,AAPL,Bloomberg.com,Apple Weighs Letting Users Switch Default iPho...,https://news.google.com/rss/articles/CBMiswFBV...
4,2020-02-05,AAPL,The Motley Fool,3 Top Large-Cap Stocks to Buy in February - Th...,https://news.google.com/rss/articles/CBMikgFBV...
...,...,...,...,...,...
11619,2026-07-08,JPM,Yahoo Finance,JPMorgan Chase (JPM) Lands $1.6 Billion Iraq U...,https://news.google.com/rss/articles/CBMilgFBV...
11620,2026-07-10,JPM,The Motley Fool,Jamie Dimon's JPMorgan Kicks Off Bank Earnings...,https://news.google.com/rss/articles/CBMimAFBV...
11621,2026-07-06,JPM,Investor's Business Daily,Bank Stocks Rally Ahead Of Earnings. Why Bulli...,https://news.google.com/rss/articles/CBMipgFBV...
11622,2026-07-07,JPM,Yahoo Finance,"JPM, BofA Eye Payments Deal — Why JPMorgan, Ba...",https://news.google.com/rss/articles/CBMilgFBV...


In [217]:
df_rss.columns

Index(['Date_Publication', 'Ticker', 'Source', 'Titre', 'URL'], dtype='object')

In [218]:
df_rss.dtypes

Date_Publication    object
Ticker              object
Source              object
Titre               object
URL                 object
dtype: object

In [219]:
df_rss.shape

(11624, 5)

In [220]:
df_rss.nunique()

Date_Publication     2003
Ticker                  9
Source                 32
Titre               10906
URL                 10923
dtype: int64

In [223]:
nombre_doublons = df_rss.duplicated(subset=['Ticker', 'Date_Publication', 'Titre']).sum()
print(f"Nombre de doublons trouvés : {nombre_doublons}")

Nombre de doublons trouvés : 0


In [221]:
df_rss['Ticker'].unique()

array(['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B',
       'JPM'], dtype=object)

In [222]:
df_rss.isnull().sum()

Date_Publication    0
Ticker              0
Source              0
Titre               0
URL                 0
dtype: int64

### 📡 Source 4 : Flux RSS Financiers (Audit des Données)

L'analyse exploratoire de cette quatrième source met en évidence un volume plus restreint mais une excellente qualité de complétude, avec une spécificité géographique ou sectorielle à noter :

*   **Volumétrie du Dataset :** Le fichier comporte **208 lignes** et **5 colonnes** (Shape: `(208, 5)`).
*   **Types de Données :** Toutes les variables (`Date_Publication`, `Ticker`, `Source`, `Titre`, `URL`) sont uniformément typées en tant qu'**objets** (`dtype='object'`).
*   **Qualité des Données :** Parfaite intégrité technique. On observe une **absence totale de valeurs manquantes** (`NaN = 0`) sur l'ensemble des colonnes.
*   **Distribution des Variables Clés :**
    *   **Tickers (6 uniques) :** Contrairement aux sources précédentes, ce dataset ne contient que **6 actifs uniques** sur les 9 de notre panel global : `['MSFT', 'AMZN', 'TSLA', 'UNH', 'BRK.B', 'JPM']`. Les tickers `AAPL`, `GOOGL` et `META` n'ont enregistré aucune entrée via ce canal sur la période de collecte.

**Conclusion pour la standardisation :** Ce sous-ensemble est techniquement très propre et prêt à l'emploi. L'absence de 3 tickers majeurs n'est pas une erreur de code, mais reflète simplement la réalité des flux RSS capturés sur cette période. Lors de la fusion globale, l'alignement se fera naturellement via l'axe des Tickers et des Dates, sans nécessiter de traitement de complétion (imputation).

## 📊 Source 5 : API Yahoo Finance

In [225]:
df_yahoo = pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\raw\yahoo_news_dataset.csv", sep=",")

In [226]:
df_yahoo

,Date_Publication,Ticker,Source,Titre,URL
0,2026-06-28,AAPL,Benzinga,"Bernie Sanders Says MacBook, iPad Price Rise A...",{'url': 'https://finance.yahoo.com/technology/...
1,2026-06-28,AAPL,Fortune,Even Apple supply chain maestro Tim Cook could...,{'url': 'https://finance.yahoo.com/technology/...
2,2026-06-28,AAPL,Insider Monkey,"Amkor Technology, Inc. (AMKR) Is A “Secret Wea...",{'url': 'https://finance.yahoo.com/markets/sto...
3,2026-06-27,AAPL,Insider Monkey,Apple (AAPL) Plans Mac Chip Roadmap Shift Towa...,{'url': 'https://finance.yahoo.com/technology/...
4,2026-06-27,AAPL,Motley Fool,The Big Paint vs Rare Earth Faceoff: One Stock...,{'url': 'https://www.fool.com/coverage/better-...
...,...,...,...,...,...
85,2026-06-27,JPM,Simply Wall St.,Centene (CNC) Appoints Lauren M. Tyler To Boar...,{'url': 'https://finance.yahoo.com/markets/sto...
86,2026-06-27,JPM,Investopedia,How Your Retirement Contributions Compare to O...,{'url': 'https://www.investopedia.com/how-your...
87,2026-06-27,JPM,Bloomberg,Tech Equity Sales Renew AI Debt-Binge Worries,{'url': 'https://finance.yahoo.com/markets/sto...
88,2026-06-27,JPM,Insider Monkey,JPMorgan Chase & Co. Names Doug Petno and Troy...,{'url': 'https://finance.yahoo.com/markets/sto...


In [227]:
df_yahoo.shape

(90, 5)

In [228]:
print(df_yahoo.columns.tolist())

['Date_Publication', 'Ticker', 'Source', 'Titre', 'URL']


In [229]:
print(df_yahoo.dtypes)

Date_Publication    object
Ticker              object
Source              object
Titre               object
URL                 object
dtype: object


In [230]:
df_yahoo.isnull().sum()

Date_Publication    0
Ticker              0
Source              0
Titre               0
URL                 0
dtype: int64

In [231]:
df_yahoo.nunique()

Date_Publication     5
Ticker               9
Source              23
Titre               75
URL                 75
dtype: int64

In [232]:
nombre_doublons = df_yahoo.duplicated(subset=['Ticker', 'Date_Publication', 'Titre']).sum()
print(f"Nombre de doublons trouvés : {nombre_doublons}")

Nombre de doublons trouvés : 0


In [97]:
df_yahoo['Ticker'].unique()

array(['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK-B',
       'JPM'], dtype=object)

### 📊 Source 5 : Yahoo Finance News (Audit des Données)

L'analyse exploratoire de cette cinquième source révèle un sous-ensemble très ciblé, techniquement sain, mais présentant des cas de redondance textuelle :

*   **Volumétrie du Dataset :** Le fichier comporte **90 lignes** et **5 colonnes** (Shape: `(90, 5)`). Il s'agit de la source la plus compacte de notre architecture.
*   **Types de Données :** L'ensemble des variables (`Date_Publication`, `Ticker`, `Source`, `Titre`, `URL`) est typé sous forme d'**objets** (`dtype='object'`).
*   **Qualité des Données :** Parfaite intégrité technique. On observe une **absence totale de valeurs manquantes** (`NaN = 0`) sur l'intégralité des colonnes.
*   **Distribution et Cardinalité (Valeurs Uniques) :**
    *   **Date_Publication (5 uniques) :** Les données proviennent d'une fenêtre temporelle très restreinte ou hautement agrégée (5 dates distinctes).
    *   **Ticker (9 uniques) :** Excellente représentativité. Malgré la faible taille du dataset, l'intégralité de nos **9 actifs de référence** est couverte.
    *   **Source (23 uniques) :** Une très forte diversité des fournisseurs d'informations au vu de la taille du fichier (23 éditeurs différents pour seulement 90 articles).
    *   **Titre & URL (75 uniques) :** Le nombre de titres uniques (75) étant inférieur au nombre de lignes (90), le fichier contient **15 doublons stricts** (articles identiques dupliqués ou associés simultanément à plusieurs tickers).

**Conclusion pour la standardisation :** Ce jeu de données partage exactement la même nomenclature de colonnes que la Source 4 (RSS) et la Source 3 (News API), ce qui facilitera grandement l'alignement. Un processus de déduplication (`drop_duplicates`) basé sur l'URL ou le Titre devra être appliqué lors de la fusion pour assainir le corpus.

## data Extraite d'apres kaggle

### 1ere source de kaggle

In [100]:
all_data = pd.read_csv(
    r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\all-data.csv", 
    sep=",", 
    encoding="latin1"
)

In [101]:
all_data

,neutral,"According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing ."
0,neutral,Technopolis plans to develop in stages an area...
1,negative,The international electronic industry company ...
2,positive,With the new production plant the company woul...
3,positive,According to the company 's updated strategy f...
4,positive,FINANCING OF ASPOCOMP 'S GROWTH Aspocomp is ag...
...,...,...
4840,negative,LONDON MarketWatch -- Share prices ended lower...
4841,neutral,Rinkuskiai 's beer sales fell by 6.5 per cent ...
4842,negative,Operating profit fell to EUR 35.4 mn from EUR ...
4843,negative,Net sales of the Paper segment decreased to EU...


In [103]:
all_data.nunique()

neutral                                                                                                                               3
According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .    4837
dtype: int64

In [105]:
print(all_data.dtypes)

neutral                                                                                                                            object
According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .    object
dtype: object


In [106]:
all_data.head()

,neutral,"According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing ."
0,neutral,Technopolis plans to develop in stages an area...
1,negative,The international electronic industry company ...
2,positive,With the new production plant the company woul...
3,positive,According to the company 's updated strategy f...
4,positive,FINANCING OF ASPOCOMP 'S GROWTH Aspocomp is ag...


In [107]:
all_data.isnull().sum()

neutral                                                                                                                            0
According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .    0
dtype: int64

===> icii il faut qu'on ajoute un colonne sentiment et tweet

### 2 eme source de Kaggle

In [110]:
data=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\data.csv", sep=",")

In [111]:
data

,SYM Im adding more every month up till the point this hits 50 SYM after that I ll start working on my SYM SYM and SYM,Bullish
0,SYM SYM SYM SYM SYM FEDs balance sheet total a...,Bearish
1,SYM SYM SYM SYM SYM X,Bullish
2,SYM going down to 300 in next 2 weeks,Bearish
3,How will this affect the market SYM price may ...,Bullish
4,SYM pos,Bullish
...,...,...
13514,SYM it s been holding pretty well at SYM 519 M...,Bearish
13515,SYM Did anyone grab and hold SYM 600 puts on M...,Bearish
13516,SYM Covered Waiting to reshort But it s ok if ...,Bearish
13517,SYM all this needs is an article that ARK or M...,Bearish


In [113]:
data.columns

Index(['SYM Im adding more every month up till the point this hits 50 SYM after that I ll start working on my SYM SYM and SYM', 'Bullish'], dtype='object')

In [112]:
data.isnull().sum()

SYM Im adding more every month up till the point this hits 50 SYM after that I ll start working on my SYM SYM and SYM    0
Bullish                                                                                                                  0
dtype: int64

In [116]:
data.shape

(13519, 2)

In [114]:
data.nunique()

SYM Im adding more every month up till the point this hits 50 SYM after that I ll start working on my SYM SYM and SYM    12519
Bullish                                                                                                                      2
dtype: int64

===> manque aussi les colonnes et il ya des doublons 

### 3 éme source 

In [234]:
stocktwits_message=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\stock_tweet_sentiment.csv", sep=",")

In [241]:
stocktwits_message.tail(5)

,Unnamed: 0,text,timestamp,source,symbols,company_names,Sentiment
28435,28435,$FB : 29234a9c-7f08-4d5a-985f-cb1a5554ecf9,Wed Jul 18 23:46:13 +0000 2018,test5f1798,FB,Facebook,0
28436,28436,【仮想通貨】ビットコインの価格上昇、８０万円台回復 約１カ月半ぶり $BTC ht...,Wed Jul 18 23:46:19 +0000 2018,keizai_toushi17,BTC,Bitcoin,0
28437,28437,RT @invest_in_hd: 'Nuff said! $TEL #telcoin #...,Wed Jul 18 23:46:19 +0000 2018,iad81,BTC,Bitcoin,0
28438,28438,【仮想通貨】ビットコインの価格上昇、８０万円台回復 約１カ月半ぶり $BTC ht...,Wed Jul 18 23:46:20 +0000 2018,O8viWMyrCV6cBOZ,BTC,Bitcoin,0
28439,28439,Stellar $XLM price: $0.297852 Binance registra...,Wed Jul 18 23:46:27 +0000 2018,Descendent92,AMP,Ameriprise Financial,0


In [239]:
stocktwits_message.shape

(28440, 7)

In [120]:
stocktwits_message.dtypes

Unnamed: 0        int64
text             object
timestamp        object
source           object
symbols          object
company_names    object
Sentiment         int64
dtype: object

In [121]:
stocktwits_message.isnull().sum()

Unnamed: 0       0
text             0
timestamp        2
source           3
symbols          3
company_names    5
Sentiment        0
dtype: int64

In [122]:
stocktwits_message.nunique()

Unnamed: 0       28440
text             25841
timestamp        26947
source            5896
symbols            454
company_names      461
Sentiment            3
dtype: int64

### 4 éme source  

In [242]:
stock_tweet_sentiment=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\stock_tweet_sentiment.csv", sep=",")

In [249]:
stock_tweet_sentiment.head()

,Unnamed: 0,text,timestamp,source,symbols,company_names,Sentiment
0,0,VIDEO: “I was in my office. I was minding my o...,Wed Jul 18 21:33:26 +0000 2018,GoldmanSachs,GS,The Goldman Sachs,0
1,1,The price of lumber $LB_F is down 22% since hi...,Wed Jul 18 22:22:47 +0000 2018,StockTwits,M,Macy's,0
2,2,Who says the American Dream is dead? https://t...,Wed Jul 18 22:32:01 +0000 2018,TheStreet,AIG,American,-1
3,3,Barry Silbert is extremely optimistic on bitco...,Wed Jul 18 22:52:52 +0000 2018,MarketWatch,BTC,Bitcoin,1
4,4,How satellites avoid attacks and space junk wh...,Wed Jul 18 23:00:01 +0000 2018,Forbes,ORCL,Oracle,-1


In [243]:
stock_tweet_sentiment.shape

(28440, 7)

### 5 éme source de Kaggle 

In [246]:
ZMSentiment=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\ZMSentiment.csv", sep=",")

In [130]:
ZMSentiment.shape

(2006, 2)

In [127]:
ZMSentiment

,SYM Got word that Portugal going to lockdown on Tuesday other countries following same means SYM usage should soar as more people have to communicate from homes I see 52 week high by EOY,Bullish
0,SYM SYM FUBO TV and ZOOM The top 2 Covid home ...,Bullish
1,SYM Its foolish buying puts especialoybon covi...,Bullish
2,SYM Biden Leads Trump by 10 points does not ma...,Bullish
3,SYM Looks like Bears are worried as the lockdo...,Bullish
4,SYM when this bubble pops it s going to be blo...,Bearish
...,...,...
2001,SYM it s been holding pretty well at SYM 519 M...,Bearish
2002,SYM Did anyone grab and hold SYM 600 puts on M...,Bearish
2003,SYM Covered Waiting to reshort But it s ok if ...,Bearish
2004,SYM all this needs is an article that ARK or M...,Bearish


In [128]:
ZMSentiment.isnull().sum()

SYM Got word that Portugal going to lockdown on Tuesday other countries following same means SYM usage should soar as more people have to communicate from homes I see 52 week high by EOY    0
Bullish                                                                                                                                                                                       0
dtype: int64

In [129]:
ZMSentiment.nunique()

SYM Got word that Portugal going to lockdown on Tuesday other countries following same means SYM usage should soar as more people have to communicate from homes I see 52 week high by EOY    1937
Bullish                                                                                                                                                                                          2
dtype: int64

==> il faut ajouter les colonnnes et ignorer les doublons alors 

### le dossier analyst 

In [255]:
analyst_ratings_processed = pd.read_csv(
    r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\analyst\analyst_ratings_processed.csv", 
    sep=",", 
    on_bad_lines="skip"
)

In [254]:
analyst_ratings_processed 

,Unnamed: 0,title,date,stock
0,0.0,Stocks That Hit 52-Week Highs On Friday,2020-06-05 10:30:00-04:00,A
1,1.0,Stocks That Hit 52-Week Highs On Wednesday,2020-06-03 10:45:00-04:00,A
2,2.0,71 Biggest Movers From Friday,2020-05-26 04:30:00-04:00,A
3,3.0,46 Stocks Moving In Friday's Mid-Day Session,2020-05-22 12:45:00-04:00,A
4,4.0,B of A Securities Maintains Neutral on Agilent...,2020-05-22 11:38:00-04:00,A
...,...,...,...,...
1400464,1413844.0,Top Narrow Based Indexes For August 29,2011-08-29 10:41:00-04:00,ZX
1400465,1413845.0,Recap: Wednesday's Top Percentage Gainers and ...,2011-06-22 16:44:00-04:00,ZX
1400466,1413846.0,UPDATE: Oppenheimer Color on China Zenix Auto ...,2011-06-21 08:26:00-04:00,ZX
1400467,1413847.0,Oppenheimer Initiates China Zenix At Outperfor...,2011-06-21 05:59:00-04:00,ZX


In [256]:
# 1. Définition de ta liste des 9 tickers cibles
tickers_recherches = ['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B', 'BRK-B', 'JPM']

# 2. Extraction des tickers uniques réellement présents dans la colonne 'stock'
tickers_presents = analyst_ratings_processed ['stock'].unique()

print("=========================================")
print("🔍 VÉRIFICATION DES 9 TICKERS CIBLES")
print("=========================================\n")

# 3. Boucle pour vérifier la présence de chaque ticker
for ticker in tickers_recherches:
    if ticker in tickers_presents:
        # Compter le nombre de lignes pour ce ticker
        nb_lignes = analyst_ratings_processed[analyst_ratings_processed['stock'] == ticker].shape[0]
        print(f"✅ {ticker} est PRÉSENT ({nb_lignes} articles trouvés)")
    else:
        print(f"❌ {ticker} est ABSENT")

print("-" * 50)
print(f"Nombre total de tickers différents dans ce fichier : {len(tickers_presents)}")

🔍 VÉRIFICATION DES 9 TICKERS CIBLES

✅ AAPL est PRÉSENT (469 articles trouvés)
❌ MSFT est ABSENT
✅ GOOGL est PRÉSENT (1585 articles trouvés)
❌ META est ABSENT
✅ AMZN est PRÉSENT (330 articles trouvés)
✅ TSLA est PRÉSENT (1930 articles trouvés)
✅ UNH est PRÉSENT (230 articles trouvés)
❌ BRK.B est ABSENT
❌ BRK-B est ABSENT
✅ JPM est PRÉSENT (10 articles trouvés)
--------------------------------------------------
Nombre total de tickers différents dans ce fichier : 6193


In [257]:
# 1. Définition des tickers à extraire
tickers_a_garder = ['AAPL', 'GOOGL', 'AMZN', 'TSLA', 'UNH', 'JPM']

# 2. Filtrage du dataset : on ne garde que les lignes où la colonne 'stock' est dans notre liste
df_analyst_filtre = analyst_ratings_processed [analyst_ratings_processed['stock'].isin(tickers_a_garder)]

# 3. Affichage du résultat
print("=========================================")
print("🎯 EXTRACTION ET FILTRAGE RÉUSSIS")
print("=========================================\n")
print(f"Dimensions du nouveau dataset filtré : {df_analyst_filtre.shape[0]} lignes | {df_analyst_filtre.shape[1]} colonnes\n")
print("Nombre de lignes conservées par entreprise :")
print(df_analyst_filtre['stock'].value_counts())

🎯 EXTRACTION ET FILTRAGE RÉUSSIS

Dimensions du nouveau dataset filtré : 4554 lignes | 4 colonnes

Nombre de lignes conservées par entreprise :
stock
TSLA     1930
GOOGL    1585
AAPL      469
AMZN      330
UNH       230
JPM        10
Name: count, dtype: int64


In [259]:
df_analyst_filtre

,Unnamed: 0,title,date,stock
3668,7120.0,Tech Stocks And FAANGS Strong Again To Start D...,2020-06-10 11:33:00-04:00,AAPL
3669,7121.0,10 Biggest Price Target Changes For Wednesday,2020-06-10 08:14:00-04:00,AAPL
3670,7122.0,"Benzinga Pro's Top 5 Stocks To Watch For Wed.,...",2020-06-10 07:53:00-04:00,AAPL
3671,7123.0,"Deutsche Bank Maintains Buy on Apple, Raises P...",2020-06-10 07:19:00-04:00,AAPL
3672,7124.0,Apple To Let Users Trade In Their Mac Computer...,2020-06-10 06:27:00-04:00,AAPL
...,...,...,...,...
1281551,1294558.0,Shares of several healthcare companies are tra...,2019-07-11 07:55:00-04:00,UNH
1281552,1294559.0,'White House kills drug rebate rule -POLITICO,2019-07-11 07:50:00-04:00,UNH
1281553,1294560.0,Healthcare Stocks Lower Ahead Of Decision From...,2019-07-09 13:39:00-04:00,UNH
1281554,1294561.0,The Daily Biotech Pulse: Roche Presents Positi...,2019-07-09 07:11:00-04:00,UNH


In [261]:
df_analyst_filtre.shape

(4554, 4)

In [302]:
df_headline=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\analyst\raw_partner_headlines.csv", sep=",")

In [303]:
df_headline.head(5)

,Unnamed: 0,headline,url,publisher,date,stock
0,2,Agilent Technologies Announces Pricing of $5……...,http://www.gurufocus.com/news/1153187/agilent-...,GuruFocus,2020-06-01 00:00:00,A
1,3,Agilent (A) Gears Up for Q2 Earnings: What's i...,http://www.zacks.com/stock/news/931205/agilent...,Zacks,2020-05-18 00:00:00,A
2,4,J.P. Morgan Asset Management Announces Liquida...,http://www.gurufocus.com/news/1138923/jp-morga...,GuruFocus,2020-05-15 00:00:00,A
3,5,"Pershing Square Capital Management, L.P. Buys ...",http://www.gurufocus.com/news/1138704/pershing...,GuruFocus,2020-05-15 00:00:00,A
4,6,Agilent Awards Trilogy Sciences with a Golden ...,http://www.gurufocus.com/news/1134012/agilent-...,GuruFocus,2020-05-12 00:00:00,A


In [304]:
# 1. Définition de ta liste des 9 tickers cibles
tickers_recherches = ['AAPL', 'MSFT', 'GOOGL', 'META', 'AMZN', 'TSLA', 'UNH', 'BRK.B', 'BRK-B', 'JPM']

# 2. Extraction des tickers uniques réellement présents dans la colonne 'stock'
tickers_presents = df_headline['stock'].unique()

print("=========================================")
print("🔍 VÉRIFICATION DES 9 TICKERS CIBLES")
print("=========================================\n")

# 3. Boucle pour vérifier la présence de chaque ticker
for ticker in tickers_recherches:
    if ticker in tickers_presents:
        # Compter le nombre de lignes pour ce ticker
        nb_lignes = df_headline[df_headline['stock'] == ticker].shape[0]
        print(f"✅ {ticker} est PRÉSENT ({nb_lignes} articles trouvés)")
    else:
        print(f"❌ {ticker} est ABSENT")

print("-" * 50)
print(f"Nombre total de tickers différents dans ce fichier : {len(tickers_presents)}")

🔍 VÉRIFICATION DES 9 TICKERS CIBLES

✅ AAPL est PRÉSENT (32 articles trouvés)
❌ MSFT est ABSENT
✅ GOOGL est PRÉSENT (175 articles trouvés)
❌ META est ABSENT
❌ AMZN est ABSENT
❌ TSLA est ABSENT
✅ UNH est PRÉSENT (170 articles trouvés)
❌ BRK.B est ABSENT
❌ BRK-B est ABSENT
✅ JPM est PRÉSENT (2873 articles trouvés)
--------------------------------------------------
Nombre total de tickers différents dans ce fichier : 6552


In [305]:
# 1. Définition des tickers à extraire
tickers_a_garder = ['AAPL', 'GOOGL','UNH', 'JPM']

# 2. Filtrage du dataset : on ne garde que les lignes où la colonne 'stock' est dans notre liste
df_headline_filtre = df_headline[df_headline['stock'].isin(tickers_a_garder)]

# 3. Affichage du résultat
print("=========================================")
print("🎯 EXTRACTION ET FILTRAGE RÉUSSIS")
print("=========================================\n")
print(f"Dimensions du nouveau dataset filtré : {df_headline_filtre.shape[0]} lignes | {df_headline_filtre.shape[1]} colonnes\n")
print("Nombre de lignes conservées par entreprise :")
print(df_headline_filtre['stock'].value_counts())

🎯 EXTRACTION ET FILTRAGE RÉUSSIS

Dimensions du nouveau dataset filtré : 3250 lignes | 6 colonnes

Nombre de lignes conservées par entreprise :
stock
JPM      2873
GOOGL     175
UNH       170
AAPL       32
Name: count, dtype: int64


In [306]:
df_headline_filtre

,Unnamed: 0,headline,url,publisher,date,stock
4067,4130,American Pie,https://talkmarkets.com/content/american-pie?p...,TalkMarkets,2020-06-02 00:00:00,AAPL
4068,4131,Tech Giants Dare Antitrust Deal Watchdogs,https://talkmarkets.com/content/tech-giants-da...,TalkMarkets,2020-06-02 00:00:00,AAPL
4069,4132,MoneyGram Shares Jump 50% As Western Union Rep...,https://talkmarkets.com/content/moneygram-shar...,TalkMarkets,2020-06-02 00:00:00,AAPL
4070,4133,All Eyes on Market Volatility,https://talkmarkets.com/content/all-eyes-on-ma...,TalkMarkets,2020-06-01 00:00:00,AAPL
4071,4134,Warren Buffett's Berkshire Hathaway Turns Up S...,http://www.gurufocus.com/news/1152965/warren-b...,GuruFocus,2020-06-01 00:00:00,AAPL
...,...,...,...,...,...,...
1680870,1684822,UnitedHealthcare Turns Cafeteria Into Meal-Pre...,http://www.gurufocus.com/news/1097788/unitedhe...,GuruFocus,2020-04-06 00:00:00,UNH
1680871,1684823,American National Bank Buys Exact Sciences Cor...,http://www.gurufocus.com/news/1097648/american...,GuruFocus,2020-04-06 00:00:00,UNH
1680872,1684824,Health Plan of Nevada and Southwest Medical As...,http://www.gurufocus.com/news/1097573/health-p...,GuruFocus,2020-04-06 00:00:00,UNH
1680873,1684825,Thinking about trading options or stock in App...,http://www.gurufocus.com/news/1097414/thinking...,GuruFocus,2020-04-06 00:00:00,UNH


### dossier stockerbot

In [ ]:
stockerbot_export = pd.read_csv(
    r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\stockerbot\stockerbot-export.csv", 
    sep=",", 
    on_bad_lines="skip"
)

In [314]:
stockerbot_export.tail()

,id,text,timestamp,source,symbols,company_names,url,verified
28259,1019730088617635800,$FB : 29234a9c-7f08-4d5a-985f-cb1a5554ecf9,Wed Jul 18 23:46:13 +0000 2018,test5f1798,FB,Facebook,NaN,False
28260,1019730115524288500,【仮想通貨】ビットコインの価格上昇、８０万円台回復 約１カ月半ぶり $BTC ht...,Wed Jul 18 23:46:19 +0000 2018,keizai_toushi17,BTC,Bitcoin,http://keizai-toushi-navi.com/?p=26838,False
28261,1019730115805184000,RT @invest_in_hd: 'Nuff said! $TEL #telcoin #...,Wed Jul 18 23:46:19 +0000 2018,iad81,BTC,Bitcoin,https://twitter.com/CRYPTOVERLOAD/status/10178...,False
28262,1019730117252341800,【仮想通貨】ビットコインの価格上昇、８０万円台回復 約１カ月半ぶり $BTC ht...,Wed Jul 18 23:46:20 +0000 2018,O8viWMyrCV6cBOZ,BTC,Bitcoin,http://true.velvet.jp/monexx/archives/2357,False
28263,1019730146180419600,Stellar $XLM price: $0.297852 Binance registra...,Wed Jul 18 23:46:27 +0000 2018,Descendent92,AMP,Ameriprise Financial,https://twitter.com/i/web/status/1019730146180...,False


In [308]:
stocks_cleaned = pd.read_csv(
    r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\stockerbot\stocks_cleaned.csv", 
    sep=",", 
    on_bad_lines="skip"
)

In [309]:
stocks_cleaned

,ticker,name
0,NVAX,Novavax
1,SIRI,Sirius XM
2,QQQ,Invesco QQQ
3,AAPL,Apple
4,MSFT,Microsoft
...,...,...
578,XYL,Xylem
579,YUM,YUM! Brands
580,ZBH,Zimmer Biomet
581,ZION,Zions Bancorporation


### dossier StockTwits_2020_2022_raw

In [316]:
df_APPL=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\StockTwits_2020_2022_Raw\AAPL_2020_2022\AAPL_188952625_188597609.csv")

In [317]:
df_APPL

,Unnamed: 0,id,body,created_at,user,source,symbols,owned_symbols,mentioned_users,entities,conversation,likes,links,reshares,reshare_message
0,0,188952625,$AAPL there is some smarty money trimming into...,2020-01-03T20:14:21Z,"{'id': 646025, 'username': 'WalterWhitesMoney'...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['9', '4', '5', '0', '5', '12', '0', '0', '0',...",[],{'sentiment': None},NaN,NaN,NaN,NaN,NaN
1,1,188952404,$AAPL boom!,2020-01-03T20:13:15Z,"{'id': 2975791, 'username': 'IntradayTradesmen...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['21', '12', '9', '0', '9', '5', '0', '4', '0'...",[],{'sentiment': None},NaN,NaN,NaN,NaN,NaN
2,2,188952295,$AAPL this pos will not fall below mid VWAP.,2020-01-03T20:12:42Z,"{'id': 2975791, 'username': 'IntradayTradesmen...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['20', '11', '9', '0', '9', '5', '0', '4', '0'...",[],{'sentiment': None},"{'parent_message_id': 188952295, 'in_reply_to_...",NaN,NaN,NaN,NaN
3,3,188952260,$AAPL I am holding going to see what this bab...,2020-01-03T20:12:31Z,"{'id': 2083522, 'username': 'robertlooney', 'n...","{'id': 2095, 'title': 'StockTwits For Android ...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['4', '4', '0', '0', '0', '3', '0', '0', '0', ...",[],{'sentiment': {'basic': 'Bullish'}},NaN,"{'total': 1, 'user_ids': [881009]}",NaN,NaN,NaN
4,4,188951817,$SPY is $AAPL over priced is the average analy...,2020-01-03T20:10:08Z,"{'id': 1581841, 'username': 'CaymanStyleBeef',...","{'id': 2095, 'title': 'StockTwits For Android ...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['3', '2', '1', '0', '1', '1', '0', '1', '0', ...",[],{'sentiment': None},"{'parent_message_id': 188951817, 'in_reply_to_...",NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3444,3444,188598126,$AAPL sold my 290 calls for 70 percent profit ...,2019-12-31T21:10:32Z,"{'id': 2888306, 'username': 'Whatismoney19', '...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['9', '3', '5', '2', '6', '1', '0', '0', '0', ...",[],{'sentiment': None},"{'parent_message_id': 188598126, 'in_reply_to_...",NaN,NaN,"{'reshared_count': 2, 'user_ids': [2888306, 28...",NaN
3445,3445,188598104,@DolanDrumpf $TSLA $SPY and $AAPL were easy pl...,2019-12-31T21:10:27Z,"{'id': 994434, 'username': 'dexteroustrader', ...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['26', '10', '14', '2', '16', '32', '0', '0', ...",['@DolanDrumpf'],{'sentiment': {'basic': 'Bullish'}},"{'parent_message_id': 188596669, 'in_reply_to_...","{'total': 1, 'user_ids': [2806635]}",NaN,NaN,NaN
3446,3446,188597945,$AAPL $SPY $NFLX $AMZN $GOOGL Happy new year ...,2019-12-31T21:09:35Z,"{'id': 1030432, 'username': 'BrianSJKim', 'nam...","{'id': 1149, 'title': 'StockTwits for iOS', 'u...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['8', '5', '3', '1', '3', '31', '0', '2', '0',...",[],{'chart': {'thumb': 'https://charts.stocktwits...,"{'parent_message_id': 188597945, 'in_reply_to_...","{'total': 18, 'user_ids': [1437561, 1058225, 1...",NaN,NaN,NaN
3447,3447,188597670,"$AAPL trading flat, then closing over yesterda...",2019-12-31T21:07:56Z,"{'id': 385977, 'username': 'TheTradingAnalyst'...","{'id': 2269, 'title': 'StockTwits Web', 'url':...","[{'id': 686, 'symbol': 'AAPL', 'title': 'Apple...","['4', '1', '3', '1', '3', '3', '0', '0', '0', ...",[],{'sentiment': None},"{'parent_message_id': 186046022, 'in_reply_to_...","{'total': 2, 'user_ids': [1623845, 201574]}","[{'title': 'Home', 'url': 'www.TheTradingAnaly...","{'reshared_count': 1, 'user_ids': [385977]}","{'reshared_count': 5, 'reshared_deleted': Fals..."


In [319]:
dossier_racine = r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\source_data_trouve_stocktwits\StockTwits_2020_2022_Raw"
tous_les_fichiers = glob.glob(os.path.join(dossier_racine, "**", "*.csv"), recursive=True)
print(f"🔍 {len(tous_les_fichiers)} fichiers CSV trouvés. Lancement de l'extraction...")

🔍 362 fichiers CSV trouvés. Lancement de l'extraction...


In [ ]:



liste_df_propres = []


for fichier in tous_les_fichiers:
    try:
        
        ticker = "INCONNU"
        if "AAPL" in fichier: ticker = "AAPL"
        elif "AMZN" in fichier: ticker = "AMZN"
        elif "FB" in fichier: ticker = "META"  
        elif "NVDA" in fichier: ticker = "NVDA"
        elif "TSLA" in fichier: ticker = "TSLA"
        
        
        df_brut = pd.read_csv(fichier)
        
       
        if 'created_at' not in df_brut.columns or 'body' not in df_brut.columns:
            continue
    
        temps_utc = pd.to_datetime(df_brut['created_at'], utc=True, errors='coerce')
        df_brut['temps_utc'] = temps_utc
        
       
        df_brut = df_brut.dropna(subset=['temps_utc'])
        
        
        df_brut = df_brut[df_brut['temps_utc'].dt.year.isin([2020, 2021, 2022])]
        
       
        if df_brut.empty:
            continue
            
      
        temps_ny = df_brut['temps_utc'].dt.tz_convert('America/New_York')
        
        
        decalage = temps_ny.apply(lambda x: pd.Timedelta(days=1) if x.hour >= 16 else pd.Timedelta(days=0))
        date_finale = temps_ny + decalage
        
       
        df_temp = pd.DataFrame()
        df_temp['Date'] = date_finale.dt.strftime('%Y-%m-%d')
        df_temp['Ticker'] = ticker
        df_temp['Titre'] = df_brut['body']
        df_temp['Source'] = 'StockTwits'
        
        
        liste_df_propres.append(df_temp)
        
    except Exception as e:
        print(f" Erreur sur le fichier {fichier} : {e}")


print("🔄 Fusion de tous les fichiers StockTwits validés...")
df_stocktwits_final = pd.concat(liste_df_propres, ignore_index=True)


df_stocktwits_final = df_stocktwits_final.drop_duplicates()

print(" TRAITEMENT TERMINÉ AVEC SUCCÈS !")
print(f" Dimensions du dataset StockTwits 2020-2022 : {df_stocktwits_final.shape}")
display(df_stocktwits_final.head())

🔄 Fusion de tous les fichiers StockTwits validés...
✅ TRAITEMENT TERMINÉ AVEC SUCCÈS !
📊 Dimensions du dataset StockTwits 2020-2022 : (4047381, 4)


,Date,Ticker,Titre,Source
0,2020-01-03,AAPL,$AAPL there is some smarty money trimming into...,StockTwits
1,2020-01-03,AAPL,$AAPL boom!,StockTwits
2,2020-01-03,AAPL,$AAPL this pos will not fall below mid VWAP.,StockTwits
3,2020-01-03,AAPL,$AAPL I am holding going to see what this bab...,StockTwits
4,2020-01-03,AAPL,$SPY is $AAPL over priced is the average analy...,StockTwits


In [323]:
# 1. On filtre le DataFrame pour garder uniquement les lignes où le Ticker n'est PAS 'NVDA'
df_stocktwits_final = df_stocktwits_final[df_stocktwits_final['Ticker'] != 'NVDA']

# 2. Vérification visuelle
print("✅ Le Ticker NVDA a été éliminé avec succès !")
print(f"📊 Nouvelles dimensions du dataset : {df_stocktwits_final.shape}")

# 3. On affiche la liste des Tickers restants pour être sûr à 100%
tickers_restants = df_stocktwits_final['Ticker'].unique()
print(f"🏷️ Tickers actuellement dans le dataset : {tickers_restants}")

✅ Le Ticker NVDA a été éliminé avec succès !
📊 Nouvelles dimensions du dataset : (3803675, 4)
🏷️ Tickers actuellement dans le dataset : ['AAPL' 'AMZN' 'META' 'TSLA']


In [324]:
df_stocktwits_final.nunique()

Date          795
Ticker          4
Titre     3449713
Source          1
dtype: int64

In [325]:
# 1. On s'assure que la colonne Date est bien comprise comme une vraie date par Pandas
df_stocktwits_final['Date'] = pd.to_datetime(df_stocktwits_final['Date'])

# 2. On extrait les informations clés
date_min = df_stocktwits_final['Date'].min().strftime('%Y-%m-%d')
date_max = df_stocktwits_final['Date'].max().strftime('%Y-%m-%d')
annees_presentes = sorted(df_stocktwits_final['Date'].dt.year.unique())

# 3. On compte le nombre de lignes par année
repartition_par_annee = df_stocktwits_final['Date'].dt.year.value_counts().sort_index()

# 4. Affichage du rapport
print("🗓️ --- RAPPORT TEMPOREL STOCKTWITS --- 🗓️")
print(f"Date la plus ancienne : {date_min}")
print(f"Date la plus récente  : {date_max}")
print(f"Années détectées      : {annees_presentes}\n")

print("📊 Répartition des messages par année :")
print(repartition_par_annee)

🗓️ --- RAPPORT TEMPOREL STOCKTWITS --- 🗓️
Date la plus ancienne : 2020-01-01
Date la plus récente  : 2022-03-05
Années détectées      : [2020, 2021, 2022]

📊 Répartition des messages par année :
Date
2020    1806042
2021    1662691
2022     334942
Name: count, dtype: int64


C:\Users\semy4\AppData\Local\Temp\ipykernel_40676\1141505718.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_stocktwits_final['Date'] = pd.to_datetime(df_stocktwits_final['Date'])


## tesla 2022

In [328]:
df_tesla=pd.read_csv(r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\raw\source_data_trouve_stocktwits\tesla_2022\stock_tweets.csv")

In [329]:
df_tesla

,Date,Tweet,Stock Name,Company Name
0,2022-09-29 23:41:16+00:00,Mainstream media has done an amazing job at br...,TSLA,"Tesla, Inc."
1,2022-09-29 23:24:43+00:00,Tesla delivery estimates are at around 364k fr...,TSLA,"Tesla, Inc."
2,2022-09-29 23:18:08+00:00,3/ Even if I include 63.0M unvested RSUs as of...,TSLA,"Tesla, Inc."
3,2022-09-29 22:40:07+00:00,@RealDanODowd @WholeMarsBlog @Tesla Hahaha why...,TSLA,"Tesla, Inc."
4,2022-09-29 22:27:05+00:00,"@RealDanODowd @Tesla Stop trying to kill kids,...",TSLA,"Tesla, Inc."
...,...,...,...,...
80788,2021-10-07 17:11:57+00:00,Some of the fastest growing tech stocks on the...,XPEV,XPeng Inc.
80789,2021-10-04 17:05:59+00:00,"With earnings on the horizon, here is a quick ...",XPEV,XPeng Inc.
80790,2021-10-01 04:43:41+00:00,Our record delivery results are a testimony of...,XPEV,XPeng Inc.
80791,2021-10-01 00:03:32+00:00,"We delivered 10,412 Smart EVs in Sep 2021, rea...",XPEV,XPeng Inc.


In [330]:
df_tesla.nunique()

Date            64424
Tweet           64479
Stock Name         25
Company Name       25
dtype: int64

In [331]:
# Extraction des valeurs uniques
tickers_uniques = df_tesla['Stock Name'].unique()

# Affichage du nombre total et de la liste
print(f"🏷️ Il y a {len(tickers_uniques)} Tickers uniques dans le dataset :")
print(tickers_uniques)

🏷️ Il y a 25 Tickers uniques dans le dataset :
['TSLA' 'MSFT' 'PG' 'META' 'AMZN' 'GOOG' 'AMD' 'AAPL' 'NFLX' 'TSM' 'KO'
 'F' 'COST' 'DIS' 'VZ' 'CRM' 'INTC' 'BA' 'BX' 'NOC' 'PYPL' 'ENPH' 'NIO'
 'ZS' 'XPEV']


In [332]:
# 2. Liste de tes Tickers cibles
tickers_cibles = ['AAPL', 'META', 'AMZN', 'MSFT', 'JPM', 'GOOGL', 'TSLA', 'BRK-B', 'UNH']

# 💡 Petite astuce : dans ta liste source on voit 'GOOG' et toi tu cherches 'GOOGL'.
# On standardise ça juste avant de filtrer pour ne pas perdre les données de Google !
df_tesla['Stock Name'] = df_tesla['Stock Name'].replace({'GOOG': 'GOOGL'})

# 3. Le filtre magique : on ne garde que les lignes où le ticker est dans ta liste
df_filtre = df_tesla[df_tesla['Stock Name'].isin(tickers_cibles)].copy()

# 4. Réinitialisation de l'index pour que ce soit propre
df_filtre = df_filtre.reset_index(drop=True)

print(f"📊 Lignes avant filtrage : {len(df_tesla)}")
print(f"🎯 Lignes après filtrage : {len(df_filtre)}")
display(df_filtre.head())

📊 Lignes avant filtrage : 80793
🎯 Lignes après filtrage : 54698


,Date,Tweet,Stock Name,Company Name
0,2022-09-29 23:41:16+00:00,Mainstream media has done an amazing job at br...,TSLA,"Tesla, Inc."
1,2022-09-29 23:24:43+00:00,Tesla delivery estimates are at around 364k fr...,TSLA,"Tesla, Inc."
2,2022-09-29 23:18:08+00:00,3/ Even if I include 63.0M unvested RSUs as of...,TSLA,"Tesla, Inc."
3,2022-09-29 22:40:07+00:00,@RealDanODowd @WholeMarsBlog @Tesla Hahaha why...,TSLA,"Tesla, Inc."
4,2022-09-29 22:27:05+00:00,"@RealDanODowd @Tesla Stop trying to kill kids,...",TSLA,"Tesla, Inc."


In [333]:
df_filtre.nunique()

Date            46295
Tweet           46328
Stock Name          6
Company Name        6
dtype: int64

In [334]:
tickers_uniques = df_tesla['Company Name'].unique()

# Affichage du nombre total et de la liste
print(f"🏷️ Il y a {len(tickers_uniques)} Tickers uniques dans le dataset :")
print(tickers_uniques)

🏷️ Il y a 25 Tickers uniques dans le dataset :
['Tesla, Inc.' 'Microsoft Corporation' 'Procter & Gamble Company'
 'Meta Platforms, Inc.' 'Amazon.com, Inc.' 'Alphabet Inc.'
 'Advanced Micro Devices, Inc.' 'Apple Inc.' 'Netflix, Inc.'
 'Taiwan Semiconductor Manufacturing Company Limited'
 'The Coca-Cola Company' 'Ford Motor Company'
 'Costco Wholesale Corporation' 'The Walt Disney Company'
 'Verizon Communications Inc.' 'Salesforce, Inc.' 'Intel Corporation'
 'The Boeing Company' 'Blackstone Inc.' 'Northrop Grumman Corporation'
 'PayPal Holdings, Inc.' 'Enphase Energy, Inc.' 'NIO Inc.' 'Zscaler, Inc.'
 'XPeng Inc.']


## 

# Fusion de dataset

In [265]:
colonnes_master = ['Date', 'Source', 'Ticker', 'Titre']
df_master = pd.DataFrame(columns=colonnes_master)

print(" Master DataFrame initialisé avec succès !")
print(f"Structure actuelle : {df_master.shape}")
display(df_master)

 Master DataFrame initialisé avec succès !
Structure actuelle : (0, 4)


,Date,Source,Ticker,Titre


In [266]:


df_temp_wsb = pd.DataFrame()
df_temp_wsb['Date'] = df_redit['Date_Publication']
df_temp_wsb['Ticker'] = df_redit['Ticker']
df_temp_wsb['Titre'] = df_redit['Titre']

df_temp_wsb['Source'] = 'Reddit_WallStreetBets'

df_master = pd.concat([df_master, df_temp_wsb], ignore_index=True)

print(f" Injection WSB réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.head())
display(df_master.tail())

 Injection WSB réussie ! Nouvelles dimensions du Master : (125, 4)


,Date,Source,Ticker,Titre
0,2026-07-10,Reddit_WallStreetBets,AAPL,Is it a sign?
1,2026-07-09,Reddit_WallStreetBets,AAPL,"Hands down, MSFT is the worst fucking investme..."
2,2026-07-08,Reddit_WallStreetBets,AAPL,Apple announces chip deal with Broadcom worth ...
3,2026-07-03,Reddit_WallStreetBets,AAPL,Every dip is buying opportunity in Apple.
4,2026-06-30,Reddit_WallStreetBets,AAPL,ASML is my first 200% gain ever. Bought during...


,Date,Source,Ticker,Titre
120,2026-03-27,Reddit_WallStreetBets,JPM,Dow Jones & NASDAQ Composite close in -10% cor...
121,2026-02-22,Reddit_WallStreetBets,JPM,The moment I buy a thousand dollars worth of puts
122,2026-01-31,Reddit_WallStreetBets,JPM,A Timeline of Today's Silver Crash and its Ben...
123,2026-01-23,Reddit_WallStreetBets,JPM,And we’re back 😏
124,2026-01-22,Reddit_WallStreetBets,JPM,Altimmune - $100k + YOLO - The MASH Sleeping G...


In [267]:
df_master.shape

(125, 4)

In [268]:



df_temp_wsb = pd.DataFrame()
df_temp_wsb['Date'] = df_redit_stocks['Date_Publication']
df_temp_wsb['Ticker'] = df_redit_stocks['Ticker']
df_temp_wsb['Titre'] = df_redit_stocks['Titre']

df_temp_wsb['Source'] = 'Reddit_Stocks'

df_master = pd.concat([df_master, df_temp_wsb], ignore_index=True)

print(f" Injection WSB réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.head())
display(df_master.tail())

 Injection WSB réussie ! Nouvelles dimensions du Master : (222, 4)


,Date,Source,Ticker,Titre
0,2026-07-10,Reddit_WallStreetBets,AAPL,Is it a sign?
1,2026-07-09,Reddit_WallStreetBets,AAPL,"Hands down, MSFT is the worst fucking investme..."
2,2026-07-08,Reddit_WallStreetBets,AAPL,Apple announces chip deal with Broadcom worth ...
3,2026-07-03,Reddit_WallStreetBets,AAPL,Every dip is buying opportunity in Apple.
4,2026-06-30,Reddit_WallStreetBets,AAPL,ASML is my first 200% gain ever. Bought during...


,Date,Source,Ticker,Titre
217,2026-05-19,Reddit_Stocks,JPM,Rste This Managed Portfolio
218,2026-05-11,Reddit_Stocks,JPM,Trading platforms
219,2026-04-24,Reddit_Stocks,JPM,The Warsh Playbook
220,2026-04-24,Reddit_Stocks,JPM,Waste Management (WM). Solid long term DCA stock?
221,2026-04-15,Reddit_Stocks,JPM,Mastercard ‘Crypto Partner Program’ March 2026


In [269]:



df_temp_news = pd.DataFrame()
df_temp_news['Date'] = df_news['Date_Publication']
df_temp_news['Ticker'] = df_news['Ticker']
df_temp_news['Source'] = df_news['Source'] 


df_news['Description'] = df_news['Description'].fillna('')


df_temp_news['Titre'] = df_news.apply(
    lambda row: f"{row['Titre']} - {row['Description']}" if str(row['Description']).strip() != '' else row['Titre'], 
    axis=1
)


df_master = pd.concat([df_master, df_temp_news], ignore_index=True)

print(f"Injection API News réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.tail())

Injection API News réussie ! Nouvelles dimensions du Master : (23926, 4)


,Date,Source,Ticker,Titre
23921,2026-07-01,Yahoo,JPM,Dimon says bull market is like a 'little tsuna...
23922,2026-07-01,Yahoo,JPM,S&P Winning Streak for July at Risk With Volat...
23923,2026-07-01,Benzinga,JPM,"'JPMorgan AI Chief, Longtime Top Trader Exits ..."
23924,2026-07-01,Yahoo,JPM,GIX Appoints Sara Dillon as Chief Technology O...
23925,2026-07-01,Yahoo,JPM,Should JPMorgan Diversified Return U.S. Small ...


In [270]:
df_master.shape

(23926, 4)

In [271]:
df_master.isnull().sum()

Date      0
Source    0
Ticker    0
Titre     0
dtype: int64

In [272]:



df_temp_wsb = pd.DataFrame()
df_temp_wsb['Date'] = df_rss['Date_Publication']
df_temp_wsb['Ticker'] = df_rss['Ticker']
df_temp_wsb['Titre'] = df_rss['Titre']
df_temp_wsb['Source'] = df_rss['Source'] 



df_master = pd.concat([df_master, df_temp_wsb], ignore_index=True)

print(f" Injection WSB réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.head())
display(df_master.tail())

 Injection WSB réussie ! Nouvelles dimensions du Master : (35550, 4)


,Date,Source,Ticker,Titre
0,2026-07-10,Reddit_WallStreetBets,AAPL,Is it a sign?
1,2026-07-09,Reddit_WallStreetBets,AAPL,"Hands down, MSFT is the worst fucking investme..."
2,2026-07-08,Reddit_WallStreetBets,AAPL,Apple announces chip deal with Broadcom worth ...
3,2026-07-03,Reddit_WallStreetBets,AAPL,Every dip is buying opportunity in Apple.
4,2026-06-30,Reddit_WallStreetBets,AAPL,ASML is my first 200% gain ever. Bought during...


,Date,Source,Ticker,Titre
35545,2026-07-08,Yahoo Finance,JPM,JPMorgan Chase (JPM) Lands $1.6 Billion Iraq U...
35546,2026-07-10,The Motley Fool,JPM,Jamie Dimon's JPMorgan Kicks Off Bank Earnings...
35547,2026-07-06,Investor's Business Daily,JPM,Bank Stocks Rally Ahead Of Earnings. Why Bulli...
35548,2026-07-07,Yahoo Finance,JPM,"JPM, BofA Eye Payments Deal — Why JPMorgan, Ba..."
35549,2026-07-07,Benzinga,JPM,JPMorgan Chase Stock Is Rising Ahead of July 1...


In [273]:



df_temp_wsb = pd.DataFrame()
df_temp_wsb['Date'] = df_yahoo['Date_Publication']
df_temp_wsb['Ticker'] = df_yahoo['Ticker']
df_temp_wsb['Titre'] = df_yahoo['Titre']
df_temp_wsb['Source'] = df_yahoo['Source'] 



df_master = pd.concat([df_master, df_temp_wsb], ignore_index=True)

print(f" Injection WSB réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.head())
display(df_master.tail())

 Injection WSB réussie ! Nouvelles dimensions du Master : (35640, 4)


,Date,Source,Ticker,Titre
0,2026-07-10,Reddit_WallStreetBets,AAPL,Is it a sign?
1,2026-07-09,Reddit_WallStreetBets,AAPL,"Hands down, MSFT is the worst fucking investme..."
2,2026-07-08,Reddit_WallStreetBets,AAPL,Apple announces chip deal with Broadcom worth ...
3,2026-07-03,Reddit_WallStreetBets,AAPL,Every dip is buying opportunity in Apple.
4,2026-06-30,Reddit_WallStreetBets,AAPL,ASML is my first 200% gain ever. Bought during...


,Date,Source,Ticker,Titre
35635,2026-06-27,Simply Wall St.,JPM,Centene (CNC) Appoints Lauren M. Tyler To Boar...
35636,2026-06-27,Investopedia,JPM,How Your Retirement Contributions Compare to O...
35637,2026-06-27,Bloomberg,JPM,Tech Equity Sales Renew AI Debt-Binge Worries
35638,2026-06-27,Insider Monkey,JPM,JPMorgan Chase & Co. Names Doug Petno and Troy...
35639,2026-06-27,Insider Monkey,JPM,JPMorgan Chase & Co. (JPM): Most Profitable Va...


In [274]:

df_temp_st = pd.DataFrame()


df_temp_st['Date'] = df_analyst_filtre['date'].astype(str).str[:10]


df_temp_st['Ticker'] = df_analyst_filtre['stock']
df_temp_st['Titre'] = df_analyst_filtre['title']


df_temp_st['Source'] = 'Benzinga'


df_master = pd.concat([df_master, df_temp_st], ignore_index=True)


print(f" Injection Benzinga réussie ! Nouvelles dimensions du Master : {df_master.shape}")
display(df_master.tail())

 Injection Benzinga réussie ! Nouvelles dimensions du Master : (40194, 4)


,Date,Source,Ticker,Titre
40189,2019-07-11,Benzinga,UNH,Shares of several healthcare companies are tra...
40190,2019-07-11,Benzinga,UNH,'White House kills drug rebate rule -POLITICO
40191,2019-07-09,Benzinga,UNH,Healthcare Stocks Lower Ahead Of Decision From...
40192,2019-07-09,Benzinga,UNH,The Daily Biotech Pulse: Roche Presents Positi...
40193,2019-07-08,Benzinga,UNH,UnitedHealthcare To Cover Inspire Medical's Sl...


In [307]:

df_temp_partenaires = pd.DataFrame()


dates_temp = pd.to_datetime(df_headline_filtre['date'])


dates_decalees = dates_temp + pd.Timedelta(days=1)


df_temp_partenaires['Date'] = dates_decalees.dt.strftime('%Y-%m-%d')


df_temp_partenaires['Ticker'] = df_headline_filtre['stock']
df_temp_partenaires['Titre'] = df_headline_filtre['headline']
df_temp_partenaires['Source'] = df_headline_filtre['publisher']

# ==========================================
# 4. FUSION DANS LE MASTER
# ==========================================
df_master = pd.concat([df_master, df_temp_partenaires], ignore_index=True)

# 5. Vérification visuelle
print("✅ Injection des Partenaires (avec décalage J+1) réussie !")
print(f"Nouvelles dimensions du Master Dataset : {df_master.shape}")
display(df_master.tail())

✅ Injection des Partenaires (avec décalage J+1) réussie !
Nouvelles dimensions du Master Dataset : (43444, 4)


,Date,Source,Ticker,Titre
43439,2020-04-07,GuruFocus,UNH,UnitedHealthcare Turns Cafeteria Into Meal-Pre...
43440,2020-04-07,GuruFocus,UNH,American National Bank Buys Exact Sciences Cor...
43441,2020-04-07,GuruFocus,UNH,Health Plan of Nevada and Southwest Medical As...
43442,2020-04-07,GuruFocus,UNH,Thinking about trading options or stock in App...
43443,2020-04-04,GuruFocus,UNH,"Stillwater Investment Management, LLC Buys SPD..."


In [315]:
# Chemin de sauvegarde sur ton bureau (avec le 'r' pour Raw String sous Windows)
chemin_sauvegarde = r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\df_master_temporaire.csv"

# Export du DataFrame en CSV
df_master.to_csv(chemin_sauvegarde, index=False, encoding='utf-8')

print("✅ Master Dataset sauvegardé avec succès pour la nuit !")

✅ Master Dataset sauvegardé avec succès pour la nuit !


In [326]:
chemin_master = r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\df_master_temporaire.csv"
df_master = pd.read_csv(chemin_master)

print(f" Dimensions du Master avant ajout : {df_master.shape}")
print(f" Dimensions de StockTwits à ajouter : {df_stocktwits_final.shape}")



 Dimensions du Master avant ajout : (43444, 4)
 Dimensions de StockTwits à ajouter : (3803675, 4)


In [ ]:

df_master_brut = pd.concat([df_master, df_stocktwits_final], ignore_index=True)

print(f"FUSION RÉUSSIE ! Dimensions du Master Brut : {df_master_brut.shape}")


chemin_final = r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\MASTER_DATASET_BRUT.csv"
df_master_brut.to_csv(chemin_final, index=False, encoding='utf-8')

print(f"Fichier brut sauvegardé dans : {chemin_final}")

FUSION RÉUSSIE ! Dimensions du Master Brut : (3847119, 4)
Fichier brut sauvegardé dans : C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\MASTER_DATASET_BRUT.csv


In [335]:
df_filtre = df_filtre.rename(columns={
    'Tweet': 'Titre',
    'Stock Name': 'Ticker'
})

# 2. On crée la colonne 'Source'
df_filtre['Source'] = 'StockTwits'

In [336]:
df_filtre['Datetime_NY'] = pd.to_datetime(df_filtre['Date'], utc=True)
# On bascule à l'heure locale de New York
df_filtre['Datetime_NY'] = df_filtre['Datetime_NY'].dt.tz_convert('America/New_York')

# Si l'heure est >= 16h (fermeture), on ajoute 1 jour, sinon on garde le même jour
df_filtre['Date_Nettoyee'] = df_filtre.apply(
    lambda row: (row['Datetime_NY'] + pd.Timedelta(days=1)).strftime('%Y-%m-%d') 
    if row['Datetime_NY'].hour >= 16 
    else row['Datetime_NY'].strftime('%Y-%m-%d'), 
    axis=1
)

# On remplace l'ancienne colonne Date par notre nouvelle date ajustée
df_filtre['Date'] = df_filtre['Date_Nettoyee']

In [337]:
df_filtre = df_filtre[['Date', 'Source', 'Ticker', 'Titre']]

In [338]:
df_filtre

,Date,Source,Ticker,Titre
0,2022-09-30,StockTwits,TSLA,Mainstream media has done an amazing job at br...
1,2022-09-30,StockTwits,TSLA,Tesla delivery estimates are at around 364k fr...
2,2022-09-30,StockTwits,TSLA,3/ Even if I include 63.0M unvested RSUs as of...
3,2022-09-30,StockTwits,TSLA,@RealDanODowd @WholeMarsBlog @Tesla Hahaha why...
4,2022-09-30,StockTwits,TSLA,"@RealDanODowd @Tesla Stop trying to kill kids,..."
...,...,...,...,...
54693,2021-09-30,StockTwits,AAPL,"All about this trendline now on $AAPL, continu..."
54694,2021-09-30,StockTwits,AAPL,This thread is just a broad overview of the ba...
54695,2021-09-30,StockTwits,AAPL,"In 2020, Tim Cook, CEO of $AAPL, earned $265,0..."
54696,2021-09-30,StockTwits,AAPL,The media is really pushing hard for their big...


In [339]:


# 5. 💥 LA FUSION ! On ajoute ces lignes ajustées à ton dataset principal (df)
df = pd.concat([df_master_brut, df_filtre], ignore_index=True)

# 6. Sécurité : on retire les doublons et on trie par date décroissante
df = df.drop_duplicates(ignore_index=True)
df['Date'] = pd.to_datetime(df['Date'], errors='coerce')
df = df.sort_values(by='Date', ascending=False)
df['Date'] = df['Date'].dt.strftime('%Y-%m-%d')
df = df.reset_index(drop=True)

print(f"🎯 FUSION ET ALIGNEMENT HORAIRE RÉUSSIS !")
print(f"📊 Nouvelles dimensions de ton Master Dataset : {df.shape}")

🎯 FUSION ET ALIGNEMENT HORAIRE RÉUSSIS !
📊 Nouvelles dimensions de ton Master Dataset : (3901772, 4)


In [341]:
chemin_final = r"C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\MASTER_DATASET_BRUT_MAX.csv"
df.to_csv(chemin_final, index=False, encoding='utf-8')

print(f"Fichier brut sauvegardé dans : {chemin_final}")

Fichier brut sauvegardé dans : C:\Users\semy4\OneDrive\Bureau\Fintech_project\data\processed\MASTER_DATASET_BRUT_MAX.csv
